tapfeatures/aux_channel_negotiator.go

```diff
@@ -0,0 +1,191 @@
+package tapfeatures
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnutils"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+// AuxChannelNegotiator is responsible for producing the extra tlv blob that is
+// encapsulated in the init and reestablish peer messages. This helps us
+// communicate custom feature bits with our peer.
+type AuxChannelNegotiator struct {
+	// peerFeatures keeps track of the supported features of our peers. This
+	// map will be used for lookups by other subsystems, when some features
+	// need to be supported by both parties to take effect.
+	peerFeatures lnutils.SyncMap[route.Vertex, *lnwire.RawFeatureVector]
+
+	// chanFeatures keeps track of the supported features of each channel.
+	// This map will be used for lookups by other subsystems to check
+	// whether certain custom channel features are supported.
+	chanFeatures lnutils.SyncMap[lnwire.ChannelID, *lnwire.RawFeatureVector]
+}
+
+// NewAuxChannelNegotiator returns a new instance of the aux channel negotiator.
+func NewAuxChannelNegotiator() *AuxChannelNegotiator {
+	return &AuxChannelNegotiator{}
+}
+
+// GetInitFeatures is called when sending an init message to a peer. It returns
+// custom feature bits to include in the init message TLVs. The implementation
+// can decide which features to advertise based on the peer's identity.
+func (n *AuxChannelNegotiator) GetInitFeatures(
+	_ route.Vertex) (tlv.Blob, error) {
```

Comment:
I think w/ the latest direction, this can now just return the feature vector directly, to be merged by lnd.